erdjs/src/smartcontracts/query.ts

```diff
@@ -127,9 +127,9 @@ export class ContractReturnData {
         this.asBuffer = Buffer.from(this.asBase64, "base64");
         this.asHex = this.asBuffer.toString("hex");
         this.asNumber = parseInt(this.asHex, 16) || 0;
-        this.asBool = this.asNumber != 0;
         this.asBigInt = new BigNumber(`0x${this.asHex || "00"}`, 16);
         this.asString = this.asBuffer.toString();
+        this.asBool = this.asNumber != 0 && this.asString !== "false" && this.asString !== "" && !this.asBigInt.eq(0, 10);
```

Comment:
so if this.asNumber == "no", this.asBool will return True :)

Suggestion to change this to:
 ```TS
this.asBool = this.asNumber != 0 && this.asString == "true" && !this.asBigInt.eq(0, 10);
```

Reply:
And maybe we should treat the strings in all case configuration. Something like:
```
this.asString.toLower !== "false"
```

Reply:
As bool should be true for everything truthy, I think "I am a helicopter" should also evaluate to `true`. If we want to go further we should lowercase `asString` so "False" will also evaluate correctly.

Reply:
Maybe we can keep the first implementation, and for strings that should evaluate to false we should have a small helper function, checking an array of the form: ["no", "0", "false"]